experiments/tf_trainer/common/model_trainer.py

```diff
@@ -287,16 +320,42 @@ def export(self, serving_input_fn, example_key_name=None):
       example includes an example_key field that is passed along by the estimator
       and returned in the predictions.
     """
+    if not metrics_key and FLAGS.n_export == -1:
+      raise ValueError(
+        'Must provide value for `metrics_key` when exporting best checkpoint.')
+
     estimator = self._estimator
     if example_key_name:
       estimator = self._add_estimator_key(self._estimator, example_key_name)
-    
+
     checkpoints_to_export = self._get_list_checkpoint(FLAGS.n_export,
                                                       self._model_dir())
+    
+    # For finding the best checkpoint.
+    best_metric = -1.0
+    best_checkpoint_path = None
 
     for checkpoint_path in checkpoints_to_export:
-      version = checkpoint_path.split('-')[-1]
-      estimator.export_savedmodel(
-          export_dir_base=os.path.join(self._model_dir(), version),
-          serving_input_receiver_fn=serving_input_fn,
-          checkpoint_path=checkpoint_path)
+      # If we are exporting the best checkpoint, see if the current checkpoint
+      # is the best.
+      if FLAGS.n_export == -1:
+        metrics = self._estimator.evaluate(self._dataset.validate_input_fn,
```

Comment:
Can you find the metric value in the checkpoint? Instead of recomputing it?

Reply:
Done, using the tfevent file now.